### PR TITLE
Add Dividends Resource

### LINKED ISSUE #19: Implement Dividends Resource
Reference: https://iextrading.com/developer/docs/#dividends

### PR DESCRIPTION
**Pull request by rodolfobandeira**
close #19

### CODE REVIEW

File: spec/iex/resources/dividends_spec.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    context 'with defaults', vcr: { cassette_name: 'dividends/msft' } do
+      subject do
+        IEX::Resources::Dividends.get('MSFT', '6m')
+      end
+      let(:dividends) { subject.first }
+      it 'retrieves dividends' do
+        expect(dividends.payment_date).to eq '2018-03-08'
+        expect(dividends.record_date).to eq '2018-02-15'
+        expect(dividends.declared_date).to eq '2017-11-29'
+        # expect(dividends.amount).to eq 0.42
```
Comment (rodolfobandeira) [inline spec/iex/resources/dividends_spec.rb]: @dblock Please, uncomment this to test the behavior I mentioned on the previous comment.

File: lib/iex/resources/dividends.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+require_relative 'base'
+
+module IEX
+  module Resources
+    class Dividends < Resource
+      property 'payment_date', from: 'paymentDate'
+      property 'record_date', from: 'recordDate'
+      property 'declared_date', from: 'declaredDate'
+      # property 'amount'
```
[the remark this answers is not in the document]
Comment (dblock) [reply lib/iex/resources/dividends.rb]: Probably by design, but definitely confusing. I opened https://github.com/intridea/hashie/pull/456.

**Comment by dblock** [comment]
Lets just rename `amount` to `total` to work around the Hashie problem and call it a day? Care to update?

File: README.md [hunk 3]
```diff
@@ -237,6 +238,24 @@ key_stats.day_5_change_percent_s # '-0.10%'
...
+
+Fetches dividends for a symbol.
+
+```ruby
+dividends = IEX::Resources::Dividends.get('MSFT', '6m') # Options are: 5y, 2y, 1y, ytd, 6m, 3m, 1m 
+
+dividends.payment_date # '2018-03-08'
+dividends.record_date # '2018-02-15'
+dividends.declared_date # '2017-11-29'
+dividends.amount_dollar # '$0.42'
```
Comment (dblock) [inline README.md]: Add `total` and `total_dollar` here.

File: lib/iex/resources/dividends.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (dblock) [reply lib/iex/resources/dividends.rb]: Change this to `property 'total', from: 'amount'`

File: lib/iex/resources/dividends.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+      property 'amount_dollar', from: 'amount', with: ->(v) { Base.to_dollar(amount: v, ignore_cents: false) }
```
Comment (dblock) [inline lib/iex/resources/dividends.rb]: Change this to `total_dollar`

**Comment by rodolfobandeira** [comment]
@dblock I saw your PR there, good catch!
I did the fixes based on your comments and also created an issue so we don't forget to add this default property `amount` back when Hashie's fix is in place.

Thank you very much!

**Comment by dblock** [comment]
There’s another workaround to just define a total_amount method without a property. Would you rather do that than having to change the api in the future?

**Comment by rodolfobandeira** [comment]
@dblock Fixed using the "temporary" `total` property creating an alias to `amount`. I would prefer always preserving the original API properties, instead of replacing with different ones. What do you think?

```ruby
alias amount total
```

**Comment by dblock** [comment]
I think you should just make a dollar method because we would otherwise need to deprecate `total` in the future. We want to have the least amount of impact on users, internal optimizations matter less.

**Comment by dblock** [comment]
Btw nice find with the trash bug, there’s now a fix in https://github.com/intridea/hashie/pull/457#pullrequestreview-145453199, and if never ceases to amaze me in how many ways Hashie can get out of control.

**Comment by rodolfobandeira** [comment]
@dblock Agree with you. Better to just live with the absence of a key rather than having to deprecate something in the future. Once the Hashie fix is released we can add the original `amount` key in here. 

Thanks!

**Comment by dblock** [comment]
Merged via https://github.com/dblock/iex-ruby-client/commit/e7e625a55e91a35eff845888ab93f99ee1df1c14. See what I did to the amount problem there, just put a workaround.

**Comment by rodolfobandeira** [comment]
Awesome, thanks @dblock !

**Pull request closed without merging**